LobbyServer2/LobbyServer/Stats/StatsApi.cs

```diff
@@ -0,0 +1,276 @@
+using System.Threading.Tasks;
+using static CentralServer.ApiServer.StatusController;
+using static EvoS.Framework.Network.Static.GameSubType;
+
+namespace CentralServer.LobbyServer.Stats
+{
+    public class StatsApi
+    {
+        private static readonly ILog log = LogManager.GetLogger(typeof(StatsApi));
+        private static StatsApi _instance;
+        private readonly StatsConfiguration conf;
+        private static HttpClient client;
+
+        public StatsApi()
+        {
+            conf = LobbyConfiguration.GetStatsConfiguration();
+            if (!conf.Enabled)
+            {
+                log.Info("StatsApi is not enabled");
+                return;
+            }
+            if (!conf.ApiUrl.StartsWith("http"))
+            {
+                log.Info("StatsApi is not configured correctly");
+                return;
+            }
+            log.Info("StatsApi Enabled");
+            // Trim the end / if its there
+            conf.ApiUrl = conf.ApiUrl.TrimEnd('/');
+            client = new HttpClient();
+            if (conf.ApiKey != null)
+            {
+                client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", conf.ApiKey);
+            }
+        }
+
+        public static StatsApi Get()
+        {
+            return _instance ??= new StatsApi();
+        }
+
+        public static PersistedAccountData GetMentorStatus(PersistedAccountData account)
+        {
+            account.Mentor = false;
+            try
+            {
+                HttpResponseMessage responseMentor = client.GetAsync($"{Get().conf.ApiUrl}/discords?filters[playername][$eq]={Uri.EscapeDataString(account.Handle)}").Result;
+                responseMentor.EnsureSuccessStatusCode();
+                string responseBody = responseMentor.Content.ReadAsStringAsync().Result;
+                JObject json = JObject.Parse(responseBody);
+                JArray dataArray = (JArray)json["data"];
+                if (dataArray != null && dataArray.Count > 0)
+                {
+                    bool mentor = dataArray[0]["attributes"]["mentor"].Value<bool>();
+                    if (mentor)
+                    {
+                        account.Mentor = true;
+                        log.Info($"Enabling Mentor status for {account.Handle}");
+                    }
+                }
+                else
+                {
+                    log.Info("No mentor information found for the given account handle.");
+                }
+            }
+            catch (HttpRequestException e)
+            {
+                log.Error($"Unable to fetch mentor status: {e.Message}");
+            }
+            catch (Exception e)
+            {
+                log.Error($"An unexpected error occurred: {e.Message}");
+            }
+            return account;
+        }
+
+        public async Task ParseStats(LobbyGameInfo gameInfo, string serverName, string serverVersion, LobbyGameSummary gameSummary)
```

Comment:
I'd rather have it wrapped in a try-catch too. Response could be invalid, and something else could go wrong. If anything happens, StatsApi should not break other subsystems.